### PR TITLE
API GW doc update to include path parameter example

### PR DESCRIPTION
**Pull request by mdeuser**

### CODE REVIEW

File: docs/apigateway.md [hunk 1]
```diff
@@ -57,21 +57,20 @@ Follow the instructions in [Configure CLI](./README.md#setting-up-the-openwhisk-
...
+  Once you get started, you will want to have full control over the HTTP response properties like `statusCode` and `headers`, and you may want to return different content types in the `body`. You can do this by using `--response-type http`, this will configure the target URL of the web action with the `http` extension.
 
   You can choose to change the code of the action to comply with the return of web actions with `http` extension or include the action in a sequence passing its result to a new action that transforms the result to be properly formatted for an HTTP response. You can read more about response types and web actions extensions in the [Web Actions](webactions.md) documentation.
 
   Change the code for the `hello.js` returning the JSON properties `body`, `statusCode` and `headers`
   ```javascript
   function main({name:name='Serverless API'}) {
       return {
-        body: new Buffer(JSON.stringify({payload:`Hello world ${name}`})).toString('base64'),
+        body: {payload:`Hello world ${name}`},
```
Comment (csantanapr) [inline docs/apigateway.md]: Thanks +1

File: docs/apigateway.md [hunk 2]
```diff
@@ -104,52 +103,61 @@ You have a series of actions to implement your backend for the book club:
 | putBooks    | PUT | updates book details |
 | deleteBooks | DELETE | deletes a book |
 
-Let's create an API for the book club, named `Book Club`, with `/club` as its HTTP URL base path and `books` as its resource.
+Let's create an API for the book club, named `Book Club`, with `/club` as its HTTP URL base path and `books` as its resource and `{isbn}` as a path parameter used to identify a specific book by its ISBN.
+
+When using path parameters, the API must be defined with a response type of `http`, and the path, including the actual path parameter value(s), will be available in the `__ow_path` field of the action's JSON parameter.
```
Comment (csantanapr) [inline docs/apigateway.md]: Add something that shows an example of what the actual value of `__ow_path` would be is not clear if it's going to include everything or just the last portion, I just tested this and it looks ls the last portion, 
__For example if the api is call with `https://${APIHOST}:9001/api/${GENERATED_API_ID}/club/books/123-abc` the value of `__ow_path` will be `/club/books/123-abc`__

File: docs/apigateway.md [hunk 3]
```diff
@@ -104,52 +103,61 @@ You have a series of actions to implement your backend for the book club:
...
+  Path: /books/{isbn}
   Verb: delete
-  URL: https://${APIHOST}:9001/api/21ef035/club/books
+  URL: https://${APIHOST}:9001/api/${GENERATED_API_ID}/club/books/{isbn}
 ```
 
 Now just for fun let's add a new book `JavaScript: The Good Parts` with a HTTP __POST__
 ```
-curl -X POST -d '{"name":"JavaScript: The Good Parts", "isbn":"978-0596517748"}' https://${APIHOST}:9001/api/21ef035/club/books
+curl -X POST -d '{"name":"JavaScript: The Good Parts", "isbn":"978-0596517748"}' -H "Content-Type: application/json" https://${APIHOST}:9001/api/${GENERATED_API_ID}/club/books
```
Comment (csantanapr) [inline docs/apigateway.md]: For creating a new book use path parameter, I think that's the value add of using path param
```
curl -X POST -d '{"name":"JavaScript: The Good Parts"}' -H "Content-Type: application/json" "https://${APIHOST}:9001/api/${GENERATED_API_ID}/club/books/978-0596517748"
```

File: docs/apigateway.md [hunk 4]
```diff
@@ -104,52 +103,61 @@ You have a series of actions to implement your backend for the book club:
...
+
+When using path parameters, the API must be defined with a response type of `http`, and the path, including the actual path parameter value(s), will be available in the `__ow_path` field of the action's JSON parameter.
 ```
-wsk api create -n "Book Club" /club /books get getBooks --response-type http
-wsk api create /club /books post postBooks              --response-type http
-wsk api create /club /books put putBooks                --response-type http
-wsk api create /club /books delete deleteBooks          --response-type http
+wsk api create -n "Book Club" /club /books/{isbn} get getBooks --response-type http
+wsk api create /club /books get getBooks                       --response-type http
+wsk api create /club /books post postBooks                     --response-type http
```
Comment (csantanapr) [inline docs/apigateway.md]: Make `post` use path param so you can use later in create new book `/books/{isbn} post postBooks`

File: docs/apigateway.md [hunk 5]
```diff
@@ -209,8 +224,9 @@ wsk api list /club
 ```
 ok: apis
 Action                    Verb         API Name        URL
-getBooks                   get         Book Club       https://${APIHOST}:9001/api/21ef035/club/books
-postBooks                 post         Book Club       https://${APIHOST}:9001/api/21ef035/club/books
-putBooks                   put         Book Club       https://${APIHOST}:9001/api/21ef035/club/books
-deleteBooks             delete         Book Club       https://${APIHOST}:9001/api/21ef035/club/books
+getBooks                   get         Book Club       https://${APIHOST}:9001/api/${GENERATED_API_ID}/club/books
+postBooks                 post         Book Club       https://${APIHOST}:9001/api/${GENERATED_API_ID}/club/books
```
Comment (csantanapr) [inline docs/apigateway.md]: same here have `post` have path param `/books/{isbn}`

File: docs/apigateway.md (hunk 2 above)
Comment (mdeuser) [reply docs/apigateway.md]: yeah.. i was not sure how to document what's being passed.  and yes, it is *not* the full url path.  i'll try to figure out a way to doc this better

File: docs/apigateway.md (hunk 5 above)
Comment (mdeuser) [reply docs/apigateway.md]: see above comment re: post + path identifier

File: docs/apigateway.md (hunk 4 above)
Comment (mdeuser) [reply docs/apigateway.md]: see above comment re: post + path identifier

File: docs/apigateway.md (hunk 3 above)
Comment (mdeuser) [reply docs/apigateway.md]: @csantanapr it's my understanding that rest apis for creating resources do not typically specify the entity identifier on the path.  instead that's either in the payload or returned in the response.

https://blog.mwaysolutions.com/2014/06/05/10-best-practices-for-better-restful-api/

**Comment by mdeuser** [comment]
@csantanapr - thanks very much for the review!

File: docs/apigateway.md (hunk 3 above)
Comment (csantanapr) [reply docs/apigateway.md]: that will teach me 🤒

File: docs/apigateway.md [hunk 6]
```diff
@@ -192,14 +205,16 @@ Now let's restore the API named `Book Club` by using the file `club-swagger.json
...
 ```
-ok: created api /books delete for action deleteBook
-https://${APIHOST}:9001/api/21ef035/club/books
-ok: created api /books get for action deleteBook
-https://${APIHOST}:9001/api/21ef035/club/books
-ok: created api /books post for action deleteBook
-https://${APIHOST}:9001/api/21ef035/club/books
-ok: created api /books put for action deleteBook
-https://${APIHOST}:9001/api/21ef035/club/books
+ok: created api /club/books/{isbn} get for action deleteBook
```
Comment (jpkessle) [inline docs/apigateway.md]: deleteBook > deleteBooks

File: docs/apigateway.md [hunk 7]
```diff
@@ -192,14 +205,16 @@ Now let's restore the API named `Book Club` by using the file `club-swagger.json
...
+https://${APIHOST}:9001/api/${GENERATED_API_ID}/club/books
+ok: created api /club/books/{isbn} put for action deleteBook
```
Comment (jpkessle) [inline docs/apigateway.md]: deleteBook > deleteBooks

File: docs/apigateway.md [hunk 8]
```diff
@@ -192,14 +205,16 @@ Now let's restore the API named `Book Club` by using the file `club-swagger.json
...
+https://${APIHOST}:9001/api/${GENERATED_API_ID}/club/books
+ok: created api /club/books/{isbn} delete for action deleteBook
```
Comment (jpkessle) [inline docs/apigateway.md]: deleteBook > deleteBooks

File: docs/apigateway.md [hunk 9]
```diff
@@ -192,14 +205,16 @@ Now let's restore the API named `Book Club` by using the file `club-swagger.json
...
+https://${APIHOST}:9001/api/${GENERATED_API_ID}/club/books
+ok: created api /club/books get for action deleteBook
```
Comment (jpkessle) [inline docs/apigateway.md]: deleteBook > deleteBooks

File: docs/apigateway.md [hunk 10]
```diff
@@ -192,14 +205,16 @@ Now let's restore the API named `Book Club` by using the file `club-swagger.json
...
+https://${APIHOST}:9001/api/${GENERATED_API_ID}/club/books
+ok: created api /club/books post for action deleteBook
```
Comment (jpkessle) [inline docs/apigateway.md]: deleteBook > deleteBooks

File: docs/apigateway.md (hunk 7 above)
Comment (mdeuser) [reply docs/apigateway.md]: indeed :+1:

**Pull request merged**